docs/the-new-architecture/pillars-fabric-components.md

```diff
@@ -109,6 +109,180 @@ Finally, we invoke the `codegenNativeComponent` generic function, passing the na
+The file shall live at the root of your Component, side-by-side to the folders created at the [folder Setup](#folder-setup) step.
+
+```json
+{
+  "name": "rnt-centered-text",
+  "version": "0.0.1",
+  "description": "Showcase a Fabric Component with a centered text",
+  "react-native": "js/index",
+  "source": "js/index",
+  "files": [
+    "js",
+    "android",
+    "ios",
+    "rnt-centered-text.podspec",
+    "!android/build",
+    "!ios/build",
+    "!**/__tests__",
+    "!**/__fixtures__",
+    "!**/__mocks__"
+  ],
+  "keywords": ["react-native", "ios", "android"],
+  "repository": "https://github.com/<your_github_handle>/rnt-centered-text",
+  "author": "<Your Name> <your_email@your_provider.com> (https://github.com/<your_github_handle>)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/<your_github_handle>/rnt-centered-text/issues"
+  },
+  "homepage": "https://github.com/<your_github_handle>/rnt-centered-text#readme",
+  "devDependencies": {},
+  "peerDependencies": {
+    "react": "*",
+    "react-native": "*"
+  },
+  // highlight-start
+  "codegenConfig": {
+    "name": "RNTCenteredTextSpecs",
+    "type": "all",
+    "jsSrcsDir": "js",
+    "android": {
+      "javaPackageName": "com.RTNCenteredText"
+    }
+  }
+  // highlight-end
+}
+```
+
+Let's break it down.
+
+The upper part of the file contains some descriptive information like the name of the Component, its version and what composes it.
+Make sure to update the various placeholders which are wrapped in `<>`, so replace all the occurrences of the `<your_github_handle>`, `<Your Name>`, and `<your_email@your_provider.com>` tokens.
+
+Then we have the dependencies for this package, specifically we need `react` and `react-native`, but you can add all the other dependencies you may have.
+
+Finally, the last important bit is the `codegenConfig` tag. This will be used **In Development** to automatically generate the code. It will be useful to see what is generated so that we can refer to the proper types. This object contains three keys:
+
+- `name`: The name of the library we are going to generate. By convention, we add the `Specs` suffix.
+- `type`: You can choose between `components` and `all`. Our suggestion is to leave `all` because it will push for extensibility in case you'll want to add also a TurboModule to this Component.
+- `jsSrcsDir`: the relative path to access the `js` specification that will be parsed by the Codegen.
+- `android`: an object that contains android-specific settings.
+  - `javaPackageName`: the name of the package that must be used to generate the code.
+
+#### iOS: Create the `podspec` file
+
+Now it's time to configure the native Component so that we can generate the required code and we can include it in our app.
+
+For iOS, we need to create a `podspec` file which will define the Component as a dependency.
+The `podspec` file for our Component will look like this
+
+```ruby title="rnt-centered-text.podspec"
```

Comment:
To be honest, the most correct naming convention would be `RTNCenteredText`. However, given that later we are going to `yarn add` the module and `yarn` lowercases the text and transform the camel-case into a Pascal-case style (dash-separated), I decided to keep this last style to avoid confusion.